flow/activities/flowable.go

```diff
@@ -297,16 +303,41 @@ func (a *FlowableActivity) MaintainPull(
 	}
 
 	done := make(chan struct{})
+	normalize := make(chan NormalizeBatchRequest)
 	a.CdcCacheRw.Lock()
 	a.CdcCache[sessionID] = CdcCacheEntry{
 		connector: srcConn,
 		done:      done,
+		normalize: normalize,
 	}
 	a.CdcCacheRw.Unlock()
 
 	ticker := time.NewTicker(15 * time.Second)
 	defer ticker.Stop()
 
+	normDone := make(chan struct{})
+	go func() {
+		for req := range normalize {
+			res, err := a.StartNormalize(ctx, &protos.StartNormalizeInput{
+				FlowConnectionConfigs: config,
+				SyncBatchID:           req.BatchID,
+			})
+			if err != nil {
+				a.Alerter.LogFlowError(ctx, config.FlowJobName, err)
```

Comment:
currently we wait until next sync batch to retry, could instead do something like
```
select {
normalize <- req:
default:
}
```
to retry unless another normalize is already queued up

Reply:
I'm an idiot, just drain channel & use goto, no need for brittle code risking self deadlock